.github/workflows/build_and_run_apps.yml

```diff
@@ -135,16 +135,27 @@ jobs:
           - runs-on: "esp32"
             marker: "generic"
             target: "esp32"
+            runs_on: [self-hosted, linux, docker, "esp32"]
+            pytest_args: ""
           - runs-on: "ESP32-ETHERNET-KIT"
             marker: "ethernet"
             target: "esp32"
+            runs_on: [self-hosted, linux, docker, "ESP32-ETHERNET-KIT"]
+            pytest_args: ""
           - runs-on: "spi_nand_flash"
             marker: "spi_nand_flash"
             target: "esp32"
+            runs_on: [self-hosted, linux, docker, "spi_nand_flash"]
+            pytest_args: ""
+          - runs-on: "qemu"
+            marker: "qemu"
+            target: "esp32s3"
+            runs_on: [self-hosted, linux, docker]
```

Comment:
I agree with the co-pilot review that we should try to avoid mixing `runs-on` and `runs_on` (hyphen vs underscore)